(AWS) cloud-controller-manager fails for nfs persistent volumes

/kind bug
/sig cloud-provider
/sig aws

**What happened**: I run the cloud-controller-manager with the PV label initializing controller inside a cluster where I use NFS volumes. Whenever creating such a PV the cloud-controller-manager panics:

```yaml
apiVersion: v1
kind: PersistentVolume
metadata:
  name: d27d7884a7700d4102b01c312d1b2450b8ab5598
spec:
  accessModes:
  - ReadWriteMany
  capacity:
    storage: 100G
  mountOptions:
  - ac
  - soft
  - intr
  - timeo=2
  - retry=5
  - retrans=3
  - lookupcache=none
  - nfsvers=3
  nfs:
    path: /exports/volumes/system-a27b6218ee/7
    server: 100.78.23.47
  persistentVolumeReclaimPolicy: Retain
  storageClassName: nfs-class
status:
  phase: Pending
```

Logs of CCM:

```
I0924 07:45:08.271581       1 controller_utils.go:1026] Caches are synced for persistent volume label controller
E0924 07:45:08.274934       1 runtime.go:66] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/workspace/anago-v1.9.10-beta.0.47+098570796b3289/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
/workspace/anago-v1.9.10-beta.0.47+098570796b3289/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/workspace/anago-v1.9.10-beta.0.47+098570796b3289/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/asm_amd64.s:509
/usr/local/go/src/runtime/panic.go:491
/usr/local/go/src/runtime/panic.go:63
/usr/local/go/src/runtime/signal_unix.go:367
/workspace/anago-v1.9.10-beta.0.47+098570796b3289/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/cloudprovider/providers/aws/aws.go:2264
/workspace/anago-v1.9.10-beta.0.47+098570796b3289/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/cloud/pvlcontroller.go:188
/workspace/anago-v1.9.10-beta.0.47+098570796b3289/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/cloud/pvlcontroller.go:180
/workspace/anago-v1.9.10-beta.0.47+098570796b3289/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/cloud/pvlcontroller.go:72
/workspace/anago-v1.9.10-beta.0.47+098570796b3289/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/cloud/pvlcontroller.go:145
/workspace/anago-v1.9.10-beta.0.47+098570796b3289/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/cloud/pvlcontroller.go:129
/workspace/anago-v1.9.10-beta.0.47+098570796b3289/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/cloud/pvlcontroller.go:119
/workspace/anago-v1.9.10-beta.0.47+098570796b3289/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/workspace/anago-v1.9.10-beta.0.47+098570796b3289/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/workspace/anago-v1.9.10-beta.0.47+098570796b3289/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:2337
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1c18e1d]

goroutine 267 [running]:
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/workspace/anago-v1.9.10-beta.0.47+098570796b3289/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x111
panic(0x413b420, 0xa74c5b0)
	/usr/local/go/src/runtime/panic.go:491 +0x283
k8s.io/kubernetes/pkg/cloudprovider/providers/aws.(*Cloud).GetLabelsForVolume(0xc42029eea0, 0xc4206a7ed0, 0xc42029eea0, 0x7f55f9564dc0, 0xc42029eea0)
	/workspace/anago-v1.9.10-beta.0.47+098570796b3289/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/cloudprovider/providers/aws/aws.go:2264 +0x2d
k8s.io/kubernetes/pkg/controller/cloud.(*PersistentVolumeLabelController).addLabelsToVolume(0xc42011da40, 0xc4206a7ed0, 0x0, 0xc4206a7ed0)
	/workspace/anago-v1.9.10-beta.0.47+098570796b3289/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/cloud/pvlcontroller.go:188 +0xb5
k8s.io/kubernetes/pkg/controller/cloud.(*PersistentVolumeLabelController).addLabels(0xc42011da40, 0xc421223ce0, 0x28, 0xc421176d88, 0xc42136d1d0)
	/workspace/anago-v1.9.10-beta.0.47+098570796b3289/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/cloud/pvlcontroller.go:180 +0x30c
k8s.io/kubernetes/pkg/controller/cloud.(*PersistentVolumeLabelController).(k8s.io/kubernetes/pkg/controller/cloud.addLabels)-fm(0xc421223ce0, 0x28, 0xc421393fa0, 0x3e3c140)
	/workspace/anago-v1.9.10-beta.0.47+098570796b3289/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/cloud/pvlcontroller.go:72 +0x3e
k8s.io/kubernetes/pkg/controller/cloud.(*PersistentVolumeLabelController).processNextWorkItem(0xc42011da40, 0x936a00)
	/workspace/anago-v1.9.10-beta.0.47+098570796b3289/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/cloud/pvlcontroller.go:145 +0xf5
k8s.io/kubernetes/pkg/controller/cloud.(*PersistentVolumeLabelController).runWorker(0xc42011da40)
	/workspace/anago-v1.9.10-beta.0.47+098570796b3289/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/cloud/pvlcontroller.go:129 +0x2b
k8s.io/kubernetes/pkg/controller/cloud.(*PersistentVolumeLabelController).(k8s.io/kubernetes/pkg/controller/cloud.runWorker)-fm()
	/workspace/anago-v1.9.10-beta.0.47+098570796b3289/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/cloud/pvlcontroller.go:119 +0x2a
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc4206174d0)
	/workspace/anago-v1.9.10-beta.0.47+098570796b3289/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x5e
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc4206174d0, 0x3b9aca00, 0x0, 0x1, 0xc420074b40)
	/workspace/anago-v1.9.10-beta.0.47+098570796b3289/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbd
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc4206174d0, 0x3b9aca00, 0xc420074b40)
	/workspace/anago-v1.9.10-beta.0.47+098570796b3289/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by k8s.io/kubernetes/pkg/controller/cloud.(*PersistentVolumeLabelController).Run
	/workspace/anago-v1.9.10-beta.0.47+098570796b3289/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/controller/cloud/pvlcontroller.go:119 +0x20e
```

**What you expected to happen**: Cloud-controller-manager only initializes volumes it is responsible for.

**How to reproduce it (as minimally and precisely as possible)**: Standard cloud-controller-manager deployment + example PV manifest above.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): `Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.10", GitCommit:"098570796b32895c38a9a1c9286425fb1ececa18", GitTreeState:"clean", BuildDate:"2018-08-02T17:11:51Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}`
- Cloud provider or hardware configuration: AWS
- OS (e.g. from /etc/os-release): CoreOS

--- a/pkg/cloudprovider/providers/aws/BUILD
+++ b/pkg/cloudprovider/providers/aws/BUILD
@@ -78,6 +78,7 @@ go_test(
     embed = [":go_default_library"],
     deps = [
         "//pkg/kubelet/apis:go_default_library",
+        "//pkg/volume:go_default_library",
         "//staging/src/k8s.io/api/core/v1:go_default_library",
         "//staging/src/k8s.io/apimachinery/pkg/apis/meta/v1:go_default_library",
         "//staging/src/k8s.io/apimachinery/pkg/types:go_default_library",

--- a/pkg/cloudprovider/providers/aws/aws.go
+++ b/pkg/cloudprovider/providers/aws/aws.go
@@ -2323,6 +2323,11 @@ func (c *Cloud) checkIfAvailable(disk *awsDisk, opName string, instance string) 
 
 // GetLabelsForVolume gets the volume labels for a volume
 func (c *Cloud) GetLabelsForVolume(ctx context.Context, pv *v1.PersistentVolume) (map[string]string, error) {
+	// Ignore if not AWSElasticBlockStore.
+	if pv.Spec.AWSElasticBlockStore == nil {
+		return nil, nil
+	}
+
 	// Ignore any volumes that are being provisioned
 	if pv.Spec.AWSElasticBlockStore.VolumeID == volume.ProvisionedVolumeName {
 		return nil, nil

--- a/pkg/cloudprovider/providers/aws/aws_test.go
+++ b/pkg/cloudprovider/providers/aws/aws_test.go
@@ -36,6 +36,7 @@ import (
 	"github.com/stretchr/testify/require"
 	"k8s.io/apimachinery/pkg/types"
 	kubeletapis "k8s.io/kubernetes/pkg/kubelet/apis"
+	"k8s.io/kubernetes/pkg/volume"
 )
 
 const TestClusterID = "clusterid.test"
@@ -905,6 +906,98 @@ func TestGetVolumeLabels(t *testing.T) {
 		kubeletapis.LabelZoneFailureDomain: "us-east-1a",
 		kubeletapis.LabelZoneRegion:        "us-east-1"}, labels)
 	awsServices.ec2.(*MockedFakeEC2).AssertExpectations(t)
+}
+
+func TestGetLabelsForVolume(t *testing.T) {
+	defaultVolume := EBSVolumeID("vol-VolumeId").awsString()
+	tests := []struct {
+		name               string
+		pv                 *v1.PersistentVolume
+		expectedVolumeID   *string
+		expectedEC2Volumes []*ec2.Volume
+		expectedLabels     map[string]string
+		expectedError      error
+	}{
+		{
+			"not an EBS volume",
+			&v1.PersistentVolume{
+				Spec: v1.PersistentVolumeSpec{},
+			},
+			nil,
+			nil,
+			nil,
+			nil,
+		},
+		{
+			"volume which is being provisioned",
+			&v1.PersistentVolume{
+				Spec: v1.PersistentVolumeSpec{
+					PersistentVolumeSource: v1.PersistentVolumeSource{
+						AWSElasticBlockStore: &v1.AWSElasticBlockStoreVolumeSource{
+							VolumeID: volume.ProvisionedVolumeName,
+						},
+					},
+				},
+			},
+			nil,
+			nil,
+			nil,
+			nil,
+		},
+		{
+			"no volumes found",
+			&v1.PersistentVolume{
+				Spec: v1.PersistentVolumeSpec{
+					PersistentVolumeSource: v1.PersistentVolumeSource{
+						AWSElasticBlockStore: &v1.AWSElasticBlockStoreVolumeSource{
+							VolumeID: "vol-VolumeId",
+						},
+					},
+				},
+			},
+			defaultVolume,
+			nil,
+			nil,
+			fmt.Errorf("no volumes found"),
+		},
+		{
+			"correct labels for volume",
+			&v1.PersistentVolume{
+				Spec: v1.PersistentVolumeSpec{
+					PersistentVolumeSource: v1.PersistentVolumeSource{
+						AWSElasticBlockStore: &v1.AWSElasticBlockStoreVolumeSource{
+							VolumeID: "vol-VolumeId",
+						},
+					},
+				},
+			},
+			defaultVolume,
+			[]*ec2.Volume{{
+				VolumeId:         defaultVolume,
+				AvailabilityZone: aws.String("us-east-1a"),
+			}},
+			map[string]string{
+				kubeletapis.LabelZoneFailureDomain: "us-east-1a",
+				kubeletapis.LabelZoneRegion:        "us-east-1",
+			},
+			nil,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			awsServices := newMockedFakeAWSServices(TestClusterID)
+			expectedVolumeRequest := &ec2.DescribeVolumesInput{VolumeIds: []*string{test.expectedVolumeID}}
+			awsServices.ec2.(*MockedFakeEC2).On("DescribeVolumes", expectedVolumeRequest).Return(test.expectedEC2Volumes)
+
+			c, err := newAWSCloud(CloudConfig{}, awsServices)
+			assert.Nil(t, err, "Error building aws cloud: %v", err)
+
+			l, err := c.GetLabelsForVolume(context.TODO(), test.pv)
+			assert.Equal(t, test.expectedLabels, l)
+			assert.Equal(t, test.expectedError, err)
+		})
+
+	}
 }
 
 func TestDescribeLoadBalancerOnDelete(t *testing.T) {

